Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -251,21 +251,23 @@ public class SQLiteBookmarksModelFactory: BookmarksModelFactory {
     }
 }
 
-private let isEditableQuery =
-    "SELECT exists( " +
-    "   SELECT exists(SELECT 1 FROM \(TableBookmarksBuffer)) AS hasBuffer, exists(SELECT 1 FROM \(TableBookmarksMirror)) AS hasMirror " +
-    "   WHERE hasBuffer == 0 OR (hasBuffer == 1 AND hasMirror == 0) " +
-    ")"
+
+private func isEditableQuery(direction: Direction) -> String {
```

Comment:
The original implementation for all of this was:

```
extension SQLiteBookmarks {
    private func getRecordsWithGUIDs(guids: [GUID], direction: Direction, includeIcon: Bool) -> Deferred<Maybe<Cursor<BookmarkNode>>> {

        let isEditable = (direction == .Local) ? 1 : 0
        let args: Args = guids.map { $0 as AnyObject }
        let varlist = BrowserDB.varlist(args.count)

        let values =
        "SELECT -1 AS id, guid, type, is_deleted, parentid, parentName, feedUri, pos, title, bmkUri, siteUri, folderName, faviconID, \(isEditable) AS isEditable " +
```

That is:
* If the direction is `.Buffer`, that means we're showing buffer items overlaid on the mirror. None of these are ever editable.
* If the direction is `.Local`, all non-folder bookmarks produced from this model are editable.

The change you made here was supposed to enhance the latter: we're editable if we're showing (local + mirror), but only if we're not out of synced — that is, we're editable if `.Local` and either the buffer is empty or the mirror is empty.

You can thus simplify this whole function:

```
private func isEditableExpression(direction: Direction) -> String {
    if direction == .Buffer {
        return "0"                     // Always false for the buffer model!
    }

    return "SELECT exists(" +
           "SELECT exists(SELECT 1 FROM \(TableBookmarksBuffer)) AS hasBuffer, exists(SELECT 1 FROM \(TableBookmarksMirror)) AS hasMirror  +
           "WHERE hasBuffer IS 0 OR hasMirror IS 0)"
}
```